framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2955,7 +2955,21 @@ private AnnotationMirror getDeclAnnotation(
 
         Set<AnnotationMirror> results = AnnotationUtils.createAnnotationSet();
         // Retrieving the annotations from the element.
-        results.addAll(elements.getAllAnnotationMirrors(elt));
+        List<? extends AnnotationMirror> fromEle = elements.getAllAnnotationMirrors(elt);
+        for (AnnotationMirror annotation : fromEle) {
+            try {
```

Comment:
The ErrorReporter throws a CheckerError that is caught by the SourceChecker. Even if we made it a warning, the Checker wouldn't continue to process the current compilation unit.